### ISSUE TITLE
[Fleet]: Agent 'Upgrade' feature fails from 7.10.1 RC Agent to 7.10.2 BC1 Kibana Cloud Build.

### ISSUE THREAD

**Issue by dikshachauhan-qasource** [elastic/beats#23462]
Kibana version:
Kibana:7.10.2 BC1 Kibana Cloud Build

Elasticsearch version:
Elasticsearch:7.10.2 BC1 Kibana Cloud Build

Host OS and Browser version:
Windows 10, All

Original install method (e.g. download page, yum, from source, etc.):
7.10.2 BC1 Kibana Cloud Build

Preconditions:
1. Artifacts for 7.10.1 Agent should be available.
	Link used: https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-7.10.1-windows-x86_64.zip
2. 7.10.2 Cloud environment should be available.

3. Agent must be deployed using default policy having system integration.

Steps to reproduce:

1. Login to Kibana cloud environment.
2. Upgrade Agent from 7.10.1 to 7.10.2 version from Fleet UI using 'upgrade Agent' option. Wait till upgrade process completes.
3. Now, navigate to Fleet>Agents Logs tab.
4. Observe, error logs generated for upgrade agent.

Reference ticket Id:
N/A

Actual Result:
Unable to upgrade 7.10.1 RC Agent on upgraded 7.10.2 BC1 Kibana Cloud Build.

Expected Result:
User should be able to upgrade 7.10.1 BC Agent on upgraded 7.10.2 Kibana Cloud Build. 

What's working:
N/A

What's not working:
N/A

Logs:
```
Application: [25637da2-e3f4-4856-95b1-edd57c41ebe9]: State changed to FAILED: failed upgrade of agent binary: 2 errors occurred: * 
package 'C:\Program Files\Elastic\Agent\data\elastic-agent-1da173\downloads\elastic-agent-7.10.2-windows-x86_64.zip' 
not found: open C:\Program Files\Elastic\Agent\data\elastic-agent-1da173\downloads\elastic-agent-7.10.2-windows-x86_64.zip: 
The system cannot find the file specified. * call to 'https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-7.10.2-windows-x86_64.zip' 
returned unsuccessful status code: 404: /go/src/github.com/elastic/beats/x-pack/elastic-agent/pkg/artifact/download/http/downloader.go[148]: unknown error
```

Screenshot:
![agent_ugrade_error_On_kibana7_10_2](https://user-images.githubusercontent.com/72375798/104181406-bd8b7d00-5434-11eb-971e-d672aaa5a027.png)

**Comment by EricDavisX** [comment]
this is merged now and waiting for a new 7.10.2 BC to test it.

**Comment by EricDavisX** [comment]
we'll need to wait for a new BC before we can test

**Comment by dikshachauhan-qasource** [comment]
Hi @EricDavisX 

We have validated this issue on 7.10.2 BC2 Kibana build and found this issue **still reproducible.**

Build details are as follows:
```
BUILD 36136
COMMIT a0b793698735eb1d0ab1038f8e5d7a951524e929
Artifact link used for 7.10.1 agent: https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-7.10.1-windows-x86_64.zip
```
Observation: Agent 'Upgrade' feature fails from 7.10.1 RC Agent to 7.10.2 BC2 Kibana Cloud Build. 

Screenshot:
![Agent_upgrade_failure](https://user-images.githubusercontent.com/72375798/104463024-45a98800-55d7-11eb-9f89-c90c1978930b.png)

Please let us know if anything else is required from our side.

Thanks
QAS

**Comment by EricDavisX** [comment]
I apologize, my recollection of the nuances around testing this feature is rusty.  Indeed we can only test this with a snapshot-kibana install if we want to use the UI feature _during testing_.  I have tested and the feature looks good to me in 7.10.2 latest Agent BC (with the fix) as tested against a 7.11 Kibana snapshot install (NOT the latest BC).  Let me explain and show evidence.  Just confirming now, if we re-test after the release is shipped it will work there too (which is the main usage).

The Agent does NOT have an easy way to know if it is running as a BC / Shipped 'stable' version or if it is running as a 'SNAPSHOT'.  We programmed the Kibana UI to always append 'SNAPSHOT' to the url location for where to look for the new agent version if the Kibana is indeed a 'SNAPSHOT' version, which is easy for Kibana to tell (not easy for Agent).  So if Kibana is a BC or stable version the url is left as is, without the 'SNAPSHOT' attached and if it is a nightly test build it will work.  this allows us to test from the UI, otherwise we couldn't test the UI at all until the product was already shipped (we can always test with the API wherein we can set the URL for the new agent as desired).

So, the summary is:
- SNAPSHOT testing during the dev cycle will work to test, we do this regularly.
- shipped 'GA' available product usage will work, because the artifacts will exist in the main elastic artifactory repo to be found. 
- BC usage will not work to test this feature because the artifactory repo DOES NOT have the soon-to-be-shipped artifact in it, but the Agent code is hard-coded to look there if Kibana does not instruct it otherwise (as Kibana does when it knows it is a snapshot).  So, we need to test before BC or during BC cycle with the latest snapshot of Kibana

Please do read thru and ask any questions, and when it is well understood we can update test content to be aware of this special need.  And I may forget it!  We can hope we will all remember the nuance to make testing smoother.  :)

Evidence of my 7.10.2 Agent and 7.11 Kibana snapshot tests:
![1-7 10 2-macSO-Agent-installed-to-7 11](https://user-images.githubusercontent.com/12970373/104471953-6fd45900-5589-11eb-8484-76b7fdd2c0ed.png)

![2-macOS-Agent-upgraded-to-7 11-ok](https://user-images.githubusercontent.com/12970373/104471952-6fd45900-5589-11eb-82c6-ae09a405b321.png)

![3-7 10 2-linux-installed-and-updating](https://user-images.githubusercontent.com/12970373/104471951-6fd45900-5589-11eb-9e93-1ebafe87a192.png)

![4-linux-upgraded-and-7 10 2-windows-installed](https://user-images.githubusercontent.com/12970373/104471950-6f3bc280-5589-11eb-961f-f0ffecab155a.png)

![5-windows-upgraded-to-7 11-successfully](https://user-images.githubusercontent.com/12970373/104471948-6f3bc280-5589-11eb-8856-0ce5a77305a9.png)


NOTE - the 'unhealthy' status of the windows host after upgrading is a separate and known problem and is believed fixed in the next BC which we will be testing. for now, i am closing this happily and we can ship the 7.10.2 release!

@dikshachauhan-qasource @rahulgupta-qasource let me know any questions, etc.  thank you

**Comment by dikshachauhan-qasource** [comment]
Hi @EricDavisX 

We have upgraded testcases as per above feedback.
Testcase link is as follows:
- [C34211](https://elastic.testrail.io/index.php?/cases/view/34211)

Please let us know if anything else is required.

Thanks
QAS
